Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -243,6 +242,10 @@ void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255
       Timer timer = get_pwm_timer(pin);
       if (timer.n == 0) return; // Don't proceed if protected timer or not recognized
       // Set compare output mode to CLEAR -> SET or SET -> CLEAR (if inverted)
+      if (timer_freq[timer.n] == 0) {           // If the timer is unconfigured and no freq is set then default PWM_FREQUENCY
+        set_pwm_frequency(pin, PWM_FREQUENCY);  // Set the frequency and save the value to the assigned index no.
+        timer_freq[timer.n] = PWM_FREQUENCY;
+      }
```

Comment:
How is it that `set_pwm_duty` would ever be called without something earlier calling `set_pwm_frequency` ? Or is this meant to also allow for clearing the stored `timer_freq` value to force re-initialization on the next call to `set_pwm_duty` ?

Reply:
It is a test to check if we need to set the timers default frequency, it works based on the fact that set_pwm_frequency is the only call that will update the value.